token/program/src/state.rs

```diff
@@ -331,9 +331,18 @@ pub trait GenericTokenAccount {
     }
 }
 
+/// The offset of state field in Account's C representation
+pub const ACCOUNT_INITIALIZED_INDEX: usize = 108;
+
+/// Check if the account data buffer represents an initialized account.
+/// This is checking the `state` (AccountState) field of an Account object.
+pub fn is_initialized_account(account_data: &[u8]) -> bool {
+    0 != *account_data.get(ACCOUNT_INITIALIZED_INDEX).unwrap_or(&0)
```

Comment:
```suggestion
    *account_data.get(ACCOUNT_INITIALIZED_INDEX).unwrap_or(&0) != 0
```
Probably just my slow brain, but I find this easier to understand with the logic flopped.

Reply:
Also, I had this thought about eliminating the magic zeroes. I know it's slightly more expensive, but might prevent confusion about the meaning of a zero someday. Wdyt? cc @joncinque 

```suggestion
    let uninitialized_account_state: u8 = AccountState::Uninitialized.into();
    *account_data.get(ACCOUNT_INITIALIZED_INDEX).unwrap_or(&uninitialized_account_state) != uninitialized_account_state
```

Or inlined:

```suggestion
    *account_data.get(ACCOUNT_INITIALIZED_INDEX).unwrap_or_else(|| &AccountState::Uninitialized.into()) != AccountState::Uninitialized as u8
```